### DISCUSSION TITLE
FrankenCoin's address used as the minter in suggestMinter

### DISCUSSION THREAD

**Discussion by dspicher** [Frankencoin-ZCHF/Frankencoin#41] [Q&A] [answered]
In the `suggestMinter()` function of the `FrankenCoin` contract, why is `address(this)` [used](https://github.com/Frankencoin-ZCHF/FrankenCoin/blob/a2ce625c554bbd3465a31e7d8b7360a054339dd2/contracts/Frankencoin.sol#L107) as the `minter` argument to `_collectProfits()` instead of the intuitively more fitting `_minter` address?

**Comment by luziusmeisser** [comment, marked as answer]
This address is only used in the Profits event. It is intended to help a frontend in figuring out which module collected how much profits. And in case of suggesting a new minter, the module that triggers the fees is the Frankencoin contract itself. Or in other words: "suggestMinter" is a feature of the Frankencoin contract and not a feature of the minter that is being suggested.

**Comment by dspicher** [comment]
That makes sense, thank you @luziusmeisser.
